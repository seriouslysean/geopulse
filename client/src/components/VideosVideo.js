import React from 'react';

const VideosVideo = ({ video }) => {
  const { id: { videoId }, snippet: { title, thumbnails: { high: image } } } = video;
  return (
    <li className="videos__video">
      <a rel="external" href={`https://www.youtube.com/watch?v=${videoId}`} target="_blank">
        <div className="videos__image" style={{ backgroundImage: `url(${image.url})` }}>
          <img src={image.url} height={image.width} width={image.width} alt={title} />
        </div>
        <div className="videos__title">{title}</div>
      </a>
    </li>
  );
};

export default VideosVideo;
