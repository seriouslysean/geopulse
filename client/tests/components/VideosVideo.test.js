import React from 'react';
import { shallow } from 'enzyme';
import VideosVideo from '../../src/components/VideosVideo';
import videosProp from '../../../data/fixtures/videos';

test('should render VideosVideo with content correctly', () => {
  const wrapper = shallow(<VideosVideo video={videosProp.collection[0]} />);
  expect(wrapper).toMatchSnapshot();
});
