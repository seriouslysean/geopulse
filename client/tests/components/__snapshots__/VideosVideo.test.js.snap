// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render VideosVideo with content correctly 1`] = `
<li
  className="videos__video"
>
  <a
    href="https://www.youtube.com/watch?v=2BsRw8wFjyI"
    rel="external"
    target="_blank"
  >
    <div
      className="videos__image"
      style={
        Object {
          "backgroundImage": "url(https://i.ytimg.com/vi/2BsRw8wFjyI/hqdefault.jpg)",
        }
      }
    >
      <img
        alt="At Strawberry Mansion High, There's Fear, Hope"
        height={480}
        src="https://i.ytimg.com/vi/2BsRw8wFjyI/hqdefault.jpg"
        width={480}
      />
    </div>
    <div
      className="videos__title"
    >
      At Strawberry Mansion High, There's Fear, Hope
    </div>
  </a>
</li>
`;
